File: Java/Programmers/prime_number_count.java
package Programmers;

public class prime_number_count {
    public int solution(int n, int k) {
        int answer = 0;
        Integer.toString(n, k);
        String num = convert(n, k) + "0";
        String tmp = "";
        for (int i = 0; i < num.length(); i ++){
            if (num.charAt(i) != '0'){
                tmp += num.charAt(i);
            } else {
                if (tmp.length() != 0){
                    if (is_prime_number(Integer.parseInt(tmp))){
                        answer += 1;
                    }
                    tmp = "";
                }
            }
        }
        return answer;
    }

    public String convert(int n, int k) {
        String result = "";
        int mod  = 0;
        while(n > 0){
            n = n / k;
            mod = n % k;
            result = Integer.toString(mod) + result;
        }
        return result;
    }

    public boolean is_prime_number(int x) {
        if(x == 1){
            return false;
        }
        for(int i = 2; i < Math.sqrt(x); i++){
            if(x%i == 0){
                return false;
            }
        }
        return true;
    }

}

File: Python/Programmers/prime_number_count.py
import math

def solution(n, k):
    answer = 0
    num = convert(n, k) + '0'
    tmp = ''
    for i in num:
        if i != '0':
            tmp += i
        else:
            if len(tmp) != 0:
                if is_prime_number(int(tmp)):
                    answer += 1
                tmp = ''
    return answer

def convert(n, k):
    result = ''
    while n > 0:
        n, mod = divmod(n, k)
        result = str(mod) + result
    return result

def is_prime_number(x):
    if x == 1:
        return False
    for i in range(2, int(math.sqrt(x) + 1)):
        if x % i == 0:
            return False
    return True